### ISSUE TITLE
E712 fix silently breaks Pandas/Polars queries

### ISSUE THREAD

**Issue by david-waterworth** [astral-sh/ruff#9883]
The following fragments

```python
df.filter(pl.col("X")==True) # polars
df[df["X"]==True] # pandas
```

are flagged as E712 violations, however the fix is invalid, and worse it silently changes the result of the query, causing subtle and hard to find errors.

```python
df.filter(pl.col("X") is True) # polars
df[df["X"] is True] # pandas
```

The correct fix is

```python
df.filter(pl.col("X")) # polars
df[df["X"]] # pandas
```

and for completeness, `==False` should be replaced by

```python
df.filter(~pl.col("X")) # polars
df[~df["X"]] # pandas
```

**Comment by charliermarsh** [comment]
I believe this fix is marked as unsafe for this reason. Did you opt-in to the fix with `--unsafe-fixes`, or do you have `unsafe-fixes = true` enabled in your project? If so, I think Ruff's behavior here is okay... I'd love to make sure we _don't_ flag these at all, but at least it's not a silent breakage, and instead one that's surfaced to the user as potentially-unsafe.

(I'm going to merge this ticket into https://github.com/astral-sh/ruff/issues/4560 which tracks the underlying limitation.)

**Comment by david-waterworth** [comment]
@charliermarsh  I think what I probably did is manually apply the fix in vscode (i.e. right click on the squiggle generated by the ruff vscode extensions) as I defn don't have `--unsafe-fixes` set.

So I guess the issue here is the extension might have displayed safe and unsafe fixes in the same colour (i.e. yellow underline) or perhaps I just didn't read the message correctly. Either way that issue should be raise in the ruff vscode extension repo so I'll double check and raise it there.

**Comment by charliermarsh** [comment]
Ahh yeah this makes total sense. I think we have an issue for that in the VS Code extension repo: https://github.com/astral-sh/ruff-vscode/issues/342.
